infrastructure/src/main/java/org/corfudb/infrastructure/log/LogMetadata.java

```diff
@@ -0,0 +1,81 @@
+package org.corfudb.infrastructure.log;
+
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.runtime.view.Address;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * A container object that holds log tail offsets and the global
+ * log tail that has been seen. Note that holes don't belong to any
+ * stream therefore the globalTail needs to be tracked separately.
+ *
+ * <p>Created by maithem on 10/15/18.
+ */
+
+@NotThreadSafe
+@ToString
+@Slf4j
+public class LogMetadata {
+
+    @Getter
+    private volatile long globalTail;
+
+    @Getter
+    final private Map<UUID, Long> streamTails;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)